### PR TITLE
Use lenses for JSON editing.  Upgrade to `hedgehog-extras0.5.1.0`

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

Upgrade to `hedgehog-extras-0.5.1.0`.  Use lenses for JSON editing.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 1]
```diff
@@ -104,18 +105,12 @@ createSPOGenesisAndFiles testnetOptions startTime (TmpAbsolutePath tempAbsPath')
...
-      . HM.insert "rho"                    (toJSON @Double 0.1)
-      . HM.insert "tau"                    (toJSON @Double 0.1)
-      . HM.insert "updateQuorum"           (toJSON @Int 2)
-      )
+  H.rewriteJsonFile createStakedInitialGenesisFile $ \o -> o
+    & L.key "securityParam" . L._Integer .~ 5
+    & L.key "rho" . L._Double  .~ 0.1
+    & L.key "tau" . L._Double  .~ 0.1
+    & L.key "updateQuorum" . L._Integer .~ 2
+    & L.key "protocolParams" . L.key "protocolVersion" . L.key "major" . L._Integer .~ 8
```
Comment (newhoggy) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:113]: It turns out that lenses make a good compatibility shim. The aeson API keeps making breaking changes (for example switching from HashMap to KeyMap lenses help us having to deal with these breaking changes.

**Comment by carbolymer** [comment]
Kind of superceded by https://github.com/IntersectMBO/cardano-node/pull/5621 ?

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 2]
```diff
@@ -104,18 +105,12 @@ createSPOGenesisAndFiles testnetOptions startTime (TmpAbsolutePath tempAbsPath')
...
-                "protocolVersion"
-          )   "protocolParams"
-      . HM.insert "rho"                    (toJSON @Double 0.1)
-      . HM.insert "tau"                    (toJSON @Double 0.1)
-      . HM.insert "updateQuorum"           (toJSON @Int 2)
-      )
+  H.rewriteJsonFile createStakedInitialGenesisFile $ \o -> o
+    & L.key "securityParam" . L._Integer .~ 5
+    & L.key "rho" . L._Double  .~ 0.1
+    & L.key "tau" . L._Double  .~ 0.1
```
Comment (carbolymer) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:111]: those are wrong, they can't be at the top level. I am going to steal your lenses idea and put it into my PR.

Comment (newhoggy) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:111]: Where should they go instead?

Comment (newhoggy) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:111]: This is not a regression in any case.

**Review by smelc — APPROVED** [review]

File: cardano-testnet/src/Testnet/Components/Configuration.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:111]: Inside `protocolParams`. This change just went overlooked, and I've stumbled upon it.

**Review by carbolymer — APPROVED** [review]

**Pull request merged**